db/src/Pos/DB/Block/Slog/Logic.hs

```diff
@@ -346,14 +382,17 @@ slogRollbackBlocks nm pc (BypassSecurityCheck bypassSecurity) (ShouldCallBListen
         map (GS.SetInMainChain False . view headerHashG) $ toList blocks
     forwardLinksBatch =
         map (GS.RemoveForwardLink . view prevBlockL) $ toList blocks
-    isGenesis0 (Left genesisBlk) = genesisBlk ^. epochIndexL == 0
-    isGenesis0 (Right _)         = False
-    lastSlotsToPrepend =
-        mapMaybe (getSlogUndo . undoSlog . snd) $ toList (toOldestFirst blunds)
-    newLastSlots lastSlots = lastSlots & _Wrapped %~ updateLastSlots
-    dropEnd n xs = take (length xs - n) xs
+
+    lastSlotsToAppend :: [LastSlotInfo]
```

Comment:
`lastSlotsToAppend` replaces `lastSlotsToPrepend` which was wrong because it failed to maintain the `OldestFirst` invariant.